src/Tests/test.e2e.ts

```diff
@@ -0,0 +1,280 @@
+import makeWASocket, { DisconnectReason, jidNormalizedUser, proto, useMultiFileAuthState } from '../index'
+import { Boom } from '@hapi/boom'
+import { readFileSync } from 'node:fs'
+import P from 'pino'
+
+jest.setTimeout(30_000)
+
+describe('E2E Tests', () => {
+	jest.useFakeTimers()
+
+	let sock: ReturnType<typeof makeWASocket>
+	let meJid: string | undefined
+
+	beforeAll(async () => {
+		const { state, saveCreds } = await useMultiFileAuthState('baileys_auth_info')
```

Comment:
The tests assume that auth already exists & that we have a stable filesystem.. We must have some sort of env variable to point to a file, and after the storage update a DB URL (which can be a simple file://sqlite.db)